In [ ]:
import numpy as np
import pandas as pd
import matplotlib as plt

## Followers

In [ ]:
df_followers = pd.read_csv('linkedin_data/followers_job.csv')
df_followers

Job function  Total followers
0                        Engineering              166
1               Business Development              155
2                           Research              117
3                          Education               80
4             Information Technology               47
5                          Marketing               47
6                         Operations               39
7                    Arts and Design               34
8                              Sales               34
9      Community and Social Services               33
10           Media and Communication               33
11    Program and Project Management               32
12                        Consulting               30
13                    Administrative               22
14                Product Management               21
15                   Human Resources               17
16                           Finance               13
17      Customer Success and Support               10
18                  Entrepreneurship                9
19                        Accounting                8
20               Healthcare Services                8
21                             Legal                8
22                        Purchasing                4
23  Military and Protective Services                3
24                 Quality Assurance                3
25                       Real Estate                2

In [ ]:
total_followers = df_followers['Total followers'].sum()

# Calculate the percentage of each 'Job function'
df_followers['Percentage'] = (df_followers['Total followers'] / total_followers) * 100

In [ ]:
df_followers.head(10)

Job function  Total followers  Percentage
0                    Engineering              166   17.025641
1           Business Development              155   15.897436
2                       Research              117   12.000000
3                      Education               80    8.205128
4         Information Technology               47    4.820513
5                      Marketing               47    4.820513
6                     Operations               39    4.000000
7                Arts and Design               34    3.487179
8                          Sales               34    3.487179
9  Community and Social Services               33    3.384615

The job functions of "Engineering" and "Business Development" have the highest total number of followers, accounting for approximately 17.03% and 15.90% of the total followers, respectively. These two categories appear to be the most popular among the LinkedIn audience in the dataset. The "Research" and "Education" job functions also demonstrate a notable presence on LinkedIn, with around 12.00% and 8.21% of the total followers, respectively. Tech-Related Functions Hold Influence: "Information Technology" and "Marketing" job functions each have around 4.82% of the total followers, highlighting the significance of tech-related and marketing roles in the LinkedIn community.

## Content Metrics

In [ ]:
df_posts = pd.read_csv('linkedin_data/posts.csv')
df_posts

Post title  \
0                                                 NaN   
1   Wondering what are latest technology trends? K...   
2                                                 NaN   
3   Business innovations are based on such applied...   
4   💜 We deeply appreciate the dedication of these...   
..                                                ...   
70  Welcome to Benjamin, who is our Lead Product D...   
71  We would like to welcome Rim, our SMM and Mark...   
72                                                NaN   
73  Please take part in our 3-5 minute survey - We...   
74  Hello from RESEARCHPRENEUERS! \n\nToday Alevti...   

                                            Post link Post type  \
0   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
1   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
2   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
3   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
4   https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
..                                                ...       ...   
70  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
71  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
72  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
73  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   
74  https://www.linkedin.com/feed/update/urn:li:ac...   Organic   

    Campaign name                         Posted by Created date  \
0             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/21/2023   
1             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/20/2023   
2             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/13/2023   
3             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/08/2023   
4             NaN  Alevtina Evgrafova, PhD, ACC ICF   07/02/2023   
..            ...                               ...          ...   
70            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/29/2022   
71            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/28/2022   
72            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/26/2022   
73            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/23/2022   
74            NaN  Alevtina Evgrafova, PhD, ACC ICF   09/21/2022   

    Campaign start date  Campaign end date       Audience  Impressions  \
0                   NaN                NaN  All followers           26   
1                   NaN                NaN  All followers          204   
2                   NaN                NaN  All followers           43   
3                   NaN                NaN  All followers          536   
4                   NaN                NaN  All followers           91   
..                  ...                ...            ...          ...   
70                  NaN                NaN  All followers          109   
71                  NaN                NaN  All followers          146   
72                  NaN                NaN  All followers          118   
73                  NaN                NaN  All followers          160   
74                  NaN                NaN  All followers          130   

    Views (Excluding offsite video views)  Offsite Views  Clicks  \
0                                     NaN            NaN       6   
1                                     NaN            NaN       5   
2                                     NaN            NaN       2   
3                                     NaN            NaN       9   
4                                     NaN            NaN       4   
..                                    ...            ...     ...   
70                                    NaN            NaN       5   
71                                    NaN            NaN       4   
72                                    NaN            NaN       8   
73                                    NaN            NaN       6   
74                                    NaN            NaN       4   

    Click through rate (CTR)  

In [ ]:
# drop repost posts
df_posts.dropna(subset=['Post title'], inplace=True)
df_posts.reset_index(drop=True, inplace=True)

In [ ]:
df_posts.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 20 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Post title                             62 non-null     object 
 1   Post link                              62 non-null     object 
 2   Post type                              62 non-null     object 
 3   Campaign name                          0 non-null      float64
 4   Posted by                              62 non-null     object 
 5   Created date                           62 non-null     object 
 6   Campaign start date                    0 non-null      float64
 7   Campaign end date                      0 non-null      float64
 8   Audience                               62 non-null     object 
 9   Impressions                            62 non-null     int64  
 10  Views (Excluding offsite video views)  16 non-null     float64
 11  Offsite 

Description:

0.	Post title: The title or headline of the LinkedIn post.
1.	Post link: The URL that directs to the LinkedIn post.
2.	Post type: The type of the post (Organic or Paid).
3.	Campaign name: The name of the advertising campaign the post was part of, if any. All the entries are NaN indicating no campaign was associated.
4.	Posted by: The name of the person or organization that posted the content.
5.	Created date: The date when the post was created.
6.	Campaign start date: The start date of the advertising campaign, if any. All the entries are NaN indicating no campaign was associated.
7.	Campaign end date: The end date of the advertising campaign, if any. All the entries are NaN indicating no campaign was associated.
8.	Audience: The audience to whom the post was targeted.
9.	Impressions: The number of times the post was shown to LinkedIn users.
10.	Views (Excluding offsite video views): The number of times the post was viewed by LinkedIn users, excluding views from offsite videos.
11.	Offsite Views: The number of views that the post got from offsite (outside of LinkedIn). All the entries are NaN suggesting no offsite views were recorded.
12.	Clicks: The number of times LinkedIn users clicked on the post.
13.	Click through rate (CTR): The percentage of impressions that resulted in a click.
14.	Likes: The number of likes the post received.
15.	Comments: The number of comments the post received.
16.	Reposts: The number of times the post was shared by LinkedIn users.
17.	Follows: The number of new followers gained as a result of the post. All the entries are NaN suggesting no new followers were gained through these posts.
18.	Engagement rate: The percentage of people who engaged with the post (through clicks, likes, comments, or shares) out of the total number of impressions.
19.	Content Type: The type of content in the post (e.g., text, image, video, etc.).

In the dataset, we have a total of 62 posts. Now, let's examine the performance of these posts. On LinkedIn, post performance can be evaluated by analyzing key metrics like impressions, engagement rate, likes, comments, and reposts. Impressions on LinkedIn refer to the number of times a post has been displayed in users' feeds. High impressions indicate that the post is being seen by a larger audience. The engagement rate on LinkedIn is calculated by comparing the total interactions (likes, comments, shares, and clicks) to the total number of impressions. A higher engagement rate signifies that the post is capturing the interest of viewers and generating meaningful interactions.

It can be seen from the chart that the majority of posts receive approximately 100 to 300, which suggests that these posts are reaching a moderate audience on LinkedIn impressions. With an engagement rate of less than 0.2, it appears that there is room for improvement in capturing the audience's interest and encouraging more interactions with the posts